Source/ASCollectionView.mm

```diff
@@ -2273,15 +2273,23 @@ - (void)didMoveToWindow
 {
   BOOL visible = (self.window != nil);
   ASDisplayNode *node = self.collectionNode;
+  BOOL rangeControllerUpdated = NO;
+
   if (!visible && node.inHierarchy) {
+    if (![node supportsRangeManagedInterfaceState]) {
+      rangeControllerUpdated = YES;
+      // Exit CellNodes first before Collection to match UIKit behaviors (tear down bottom up).
+      // Although we have not yet cleared the interfaceState's Visible bit (this  happens in __exitHierarchy),
+      // the ASRangeController will get the correct value from -interfaceStateForRangeController:.
+      [_rangeController updateRanges];
```

Comment:
Normally visible cells will exit first before collection in UIKit; 
This change is to ask cells in preload/visible state to exit "together" with visible ones (before collection exit)